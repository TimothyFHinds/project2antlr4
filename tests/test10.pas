program test10;
var
   a, b, c: integer;

procedure display();
var
   a, b, c: integer;
begin
   (* local variables *)
   a := 10;
   b := 20;
   c := a + b;
   
   writeln('Within the procedure display');
   writeln('value of a = ');
   writeln(a);
   writeln('value of b = ' );
   writeln(b);
   writeln('value of c = ' );
   writeln(c);

end;

begin
   a:= 100;
   b:= 200;
   c:= a + b;
   
   writeln('Within the program');
   writeln('value of a = ');
   writeln(a);
   writeln('value of b = ' );
   writeln(b);
   writeln('value of c = ' );
   writeln(c);   
   display();
end.