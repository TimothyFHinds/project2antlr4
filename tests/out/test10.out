'Within the program'
'value of a = '
100.0
'value of b = '
200.0
'value of c = '
300.0
ALREADY A GLOBAL VARIABLE!
ALREADY A GLOBAL VARIABLE!
ALREADY A GLOBAL VARIABLE!
'Within the procedure display'
'value of a = '
10.0
'value of b = '
20.0
'value of c = '
30.0
